### PR TITLE
More safety against missing data in firestore

### PR DESCRIPTION
**Pull request by thecadams**

### CODE REVIEW

**Comment by thecadams** [comment]
> Code Climate has analyzed commit [c3937a7](https://github.com/sparkletown/sparkle/commit/c3937a70b9f14246dcf924a7b3101a015093e85b) and detected **2 issues** on this pull request.
> ...
> Duplication	2

Seems minor; I'm going to ignore this. We only added this today, so it's still being calibrated, but to my mind this type of duplication is far beneath the level we are concerned about.

File: src/pages/Admin/Admin.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ const VenueList: React.FC<VenueListProps> = ({
             <Link to={`/admin/venue/${venue.id}`}>{venue.name}</Link>
             {isVenueWithRooms(venue) && (
               <ul className="page-container-adminsidebar-subvenueslist">
-                {venue.rooms.map((room, idx) => (
+                {venue.rooms?.map((room, idx) => (
```
Comment (thecadams) [inline src/pages/Admin/Admin.tsx:105]: This is the issue that was crashing the Admin page. I thought it'd be prudent to fix all the rest of the trouble with the PartyMap template.

Comment (denisdimitrov) [reply src/pages/Admin/Admin.tsx:105]: I suppose our Types are incorrect and give false sense of safety. We should probably have a call and discuss this.

Comment (denisdimitrov) [reply src/pages/Admin/Admin.tsx:105]: This could be an issue in other components as well, at this very moment.

Comment (0xdevalias) [reply src/pages/Admin/Admin.tsx:105]: https://github.com/sparkletown/sparkle/pull/744 my 'we should fix the types' PR that I keep not having time to get back to.

File: src/types/PartyMapVenue.ts [hunk 2]
```diff
@@ -8,24 +8,24 @@ export const isPartyMapVenue = (venue: Venue): venue is PartyMapVenue =>
...
+  unhosted_entry_video_url?: string;
+  map_url?: string;
+  map_viewbox?: string;
+  password?: string;
   admin_password?: string;
-  owners: string[];
-  rooms: PartyMapRoomData[];
+  owners?: string[];
+  rooms?: PartyMapRoomData[];
 }
```
Comment (0xdevalias) [inline src/types/PartyMapVenue.ts:31]: 👌 🎉

**Review by 0xdevalias — APPROVED** [review]
[CRO] Changes look good, LGTM!

**Comment by 0xdevalias** [comment]
> @thecadams Seems minor; I'm going to ignore this. We only added this today, so it's still being calibrated, but to my mind this type of duplication is far beneath the level we are concerned about.

Yup, agreed. I haven't set it up to block PR's, so we can treat it as guidance/FYI, 'providing info and giving us a feel for what we want to calibrate/change in it, etc.

**Pull request merged**